cmd/atlas/internal/cmdext/cmdext.go

```diff
@@ -459,3 +461,91 @@ func dirFormatter(dir migrate.Dir) migrate.Formatter {
 		return migrate.DefaultFormatter
 	}
 }
+
+// ArchiveDir returns a tar archive of the given directory.
+func ArchiveDir(dir migrate.Dir) ([]byte, error) {
+	var buf bytes.Buffer
+	tw := tar.NewWriter(&buf)
+	defer tw.Close()
+
+	sumf, err := dir.Checksum()
+	if err != nil {
+		return nil, err
+	}
+	sumt, err := sumf.MarshalText()
+	if err != nil {
+		return nil, err
+	}
+	if err := append2Tar(tw, migrate.HashFileName, sumt); err != nil {
+		return nil, err
+	}
+	files, err := dir.Files()
+	if err != nil {
+		return nil, err
+	}
+	for _, f := range files {
+		if err := append2Tar(tw, f.Name(), f.Bytes()); err != nil {
+			return nil, err
+		}
+	}
+	return buf.Bytes(), nil
+}
+
+// UnarchiveDir extracts the tar archive into the given directory. If the archive contains
+// a file named "atlas.sum", it will be used to verify the checksum of the directory.
+func UnarchiveDir(arc []byte, dir migrate.Dir) error {
```

Comment:
I would say, change the API to `UnarchiveDir([]byte) (migrate.Dir, err)` and always return MemDir, because in case of an error (below), the LocalDir will end up in a stale state.